From 48493e305cfd78a6b823cc92e916d8b79340877d Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Thu, 7 Apr 2022 11:02:29 -0700
Subject: [PATCH 14/14] BUG Fix signature of D_IIR_forback(1,2)

The precision parameter has type float but it is declared in `_splinemodule.c`
with type double.
---
 scipy/signal/_splinemodule.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scipy/signal/_splinemodule.c b/scipy/signal/_splinemodule.c
index f66dd79aa..d4bcc1460 100644
--- a/scipy/signal/_splinemodule.c
+++ b/scipy/signal/_splinemodule.c
@@ -15,14 +15,14 @@ extern int S_separable_2Dconvolve_mirror(float*,float*,int,int,float*,float*,int
 
 extern int D_cubic_spline2D(double*,double*,int,int,double,npy_intp*,npy_intp*,double);
 extern int D_quadratic_spline2D(double*,double*,int,int,double,npy_intp*,npy_intp*,double);
-extern int D_IIR_forback1(double,double,double*,double*,int,int,int,double);
-extern int D_IIR_forback2(double,double,double*,double*,int,int,int,double);
+extern int D_IIR_forback1(double,double,double*,double*,int,int,int,float);
+extern int D_IIR_forback2(double,double,double*,double*,int,int,int,float);
 extern int D_separable_2Dconvolve_mirror(double*,double*,int,int,double*,double*,int,int,npy_intp*,npy_intp*);
 
 #ifdef __GNUC__
 extern int C_IIR_forback1(__complex__ float,__complex__ float,__complex__ float*,__complex__ float*,int,int,int,float);
 extern int C_separable_2Dconvolve_mirror(__complex__ float*,__complex__ float*,int,int,__complex__ float*,__complex__ float*,int,int,npy_intp*,npy_intp*);
-extern int Z_IIR_forback1(__complex__ double,__complex__ double,__complex__ double*,__complex__ double*,int,int,int,double);
+extern int Z_IIR_forback1(__complex__ double,__complex__ double,__complex__ double*,__complex__ double*,int,int,int,float);
 extern int Z_separable_2Dconvolve_mirror(__complex__ double*,__complex__ double*,int,int,__complex__ double*,__complex__ double*,int,int,npy_intp*,npy_intp*);
 #endif
 
-- 
2.25.1

